enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,400 @@
+
+## Summary
+
+This enhancement proposes a mechanism for blocking edges for the subset of clusters considered vulnerable to known issues with a particular update or target release.
+
+## Motivation
+
+When managing the [Cincinnati][cincinnati-spec] update graph [for OpenShift][cincinnati-for-openshift-design], we sometimes discover issues with particular release images or updates between them.
+Once an issue is discovered, we [block edges][block-edges] so we no longer recommend risky updates, or updates to risky releases.
+
+Note: as described [in the documentation][support-documentation], supported updates are still supported even if incoming edges are blocked, and Red Hat will eventually provide supported update paths from any supported release to the latest supported release in its z-stream.
+And, since [docs#32091][openshift-docs-32091], [that documentation][support-documentation] also points out that updates initiated after the update recommendation has been removed are still supported.
+
+Incoming bugs are evaluated to determine an impact statement based on [a generic template][rhbz-1858026-impact-statement-request].
+Some bugs only impact specific platforms, or clusters with other specific features.
+For example, rhbz#1858026 [only impacted][rhbz-1858026-impact-statement] clusters with the `None` platform which were created as 4.1 clusters and subsequently updated via 4.2, 4.3, and 4.4 to reach 4.5.
+And rhbz#1957584 [only impacted][rhbz-1957584-impact-statement] clusters updating from 4.6 to 4.7 with Routes whose `spec.host` contains no dots or was otherwise an invalid domain name.
+
+In those cases there is currently tension between wanting to protect vulnerable clusters by blocking the edge vs. wanting to avoid inconveniencing clusters which we know are not vulnerable and whose administrators may have been planning on taking the recommended update.
+This enhancement aims to reduce that tension.
+
+### Goals
+
+* [Cincinnati graph-data][graph-data] maintainers will have the ability to block edges for a vulnerable subset of clusters.
+
+### Non-Goals
+
+* Exactly scoping the set of blocked clusters to those which would have been impacted by the issue.
+    For example, some issues may be races where the impacted cluster set is a random subset of the vulnerable cluster set.
+    Any targeting of the blocked edges will reduce the number of blocked clusters which would have not been impacted, and thus reduce the contention between protecting vulnerable clusters and inconveniencing invulnerable clusters.
+* Specifying a particular update service implementation.
+    This enhancement floats some ideas, but the details of the chosen approach are up to each update service's maintainers.
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The blocked-edges schema][block-edges] will be extended with the following new properties:
```

Comment:
why did we decide this belongs in the "blocked edges" schema and not the "edges" schema?  it's slightly odd that they are defined as "conditionally blocked edges" here, but then get translated into "conditionally allowed edges" in the next section.

Reply:
(i'm guessing this is tied to wanting the default behavior of older clients to be that it treats this content as always blocked, since it can't parse/understand the conditions, so maybe we can't reconcile the two)

Reply:
We already have https://github.com/openshift/cincinnati-graph-data/tree/master/blocked-edges in in practice. Which Cincinnati graph builder understands. @wking you might want to explictly mention what we have in production.

Reply:
Yeah, there is no "edges schema" in graph-data.  I tried to sell folks on making edges a first-class graph-data concept explicit in #1, but lost out to loading edges dynamically from release-image metadata.  I'll reroll to include that context.

Reply:
@bparees Does the additional context help or does this still seem worth considering other options?

Reply:
it helps.